File: src/main/java/net/ccbluex/liquidbounce/utils/timer/TimeUtils.java
package net.ccbluex.liquidbounce.utils.timer;


import net.ccbluex.liquidbounce.utils.math.MathUtils;
import net.ccbluex.liquidbounce.utils.misc.RandomUtils;

public final class TimeUtils {

    public static long randomDelay(final int minDelay, final int maxDelay) {
        return RandomUtils.nextInt(minDelay, maxDelay);
    }

    public static long randomClickDelay(final int minCPS, final int maxCPS) {
        return (long) ((Math.random() * (1000 / minCPS - 1000 / maxCPS) * 1.03 + MathUtils.randomNumber(3, 1)) + 1050 / maxCPS);
    }
}

File: src/main/java/net/ccbluex/liquidbounce/utils/timer/TimeUtils.kt
// LiquidX Development by PrahXZ and Haflin with FDP Base modified. v2.0 R1
package net.ccbluex.liquidbounce.utils.timer

import net.ccbluex.liquidbounce.utils.misc.RandomUtils

object TimeUtils {
    fun randomDelay(minDelay: Int, maxDelay: Int): Long {
        val baseDelay = RandomUtils.nextInt(minDelay, maxDelay).toLong()
        val randomOffset = RandomUtils.nextInt(0, 20).toLong()
        return baseDelay + randomOffset
    }
    fun randomClickDelay(minCPS: Int, maxCPS: Int): Long {
        val baseDelay = (Math.random() * (1000 / minCPS - 1000 / maxCPS + 1) + 1000 / maxCPS).toLong()
        val randomOffset = RandomUtils.nextInt(0, 20).toLong()
        return baseDelay + randomOffset
    }
}